### PR TITLE
Make StandardAssetLoader public/unstable.

### PR DESCRIPTION
**Pull request by grokys**
## What does the pull request do?

Undoes the change in #11190 to make `StandardAssetLoader` internal. Make it public/unstable along with related interfaces (concrete implementations currently remain internal). This allows registration of a `StandardAssetLoader` when bypassing reference assemblies.

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0034133-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0034141-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0034221-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: src/Avalonia.Base/Platform/Internal/AssemblyDescriptor.cs [hunk 1]
```diff
@@ -3,11 +3,13 @@
...
 using System.Linq;
 using System.Reflection;
+using Avalonia.Metadata;
 using Avalonia.Utilities;
 
 namespace Avalonia.Platform.Internal;
 
-internal interface IAssemblyDescriptor
+[Unstable]
+public interface IAssemblyDescriptor
```
Comment (maxkatz6) [inline src/Avalonia.Base/Platform/Internal/AssemblyDescriptor.cs:12]: These interfaces should be still internal. They always have been internal.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0034282-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: src/Avalonia.Base/Platform/Internal/AssemblyDescriptor.cs (hunk 1 above)
Comment (Gillibald) [reply src/Avalonia.Base/Platform/Internal/AssemblyDescriptor.cs:12]: I looked at the codebase and it seems members marked with unstable are still public. They will be hidden in the ref assembly so it should be fine.

**Review by maxkatz6 — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0034348-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Pull request merged**